s3/src/main/scala/uk/gov/nationalarchives/DAS3Client.scala

```diff
@@ -163,11 +113,87 @@ object DAS3Client:
     .s3Client(asyncClient)
     .build()
 
-  def apply[F[_]: Async](): DAS3Client[F] = new DAS3Client[F](transferManager, asyncClient)
-
-  def apply[F[_]: Async](asyncClient: S3AsyncClient): DAS3Client[F] =
-    val transferManager: S3TransferManager = S3TransferManager
-      .builder()
-      .s3Client(asyncClient)
-      .build()
-    new DAS3Client[F](transferManager, asyncClient)
+  def apply[F[_]: Async](): DAS3Client[F] = DAS3Client(transferManager, asyncClient)
+
+  def apply[F[_]: Async](asyncClient: S3AsyncClient): DAS3Client[F] = DAS3Client(transferManager, asyncClient)
+
+  def apply[F[_]: Async](transferManager: S3TransferManager, asyncClient: S3AsyncClient): DAS3Client[F] =
```

Comment:
Instead of the `apply` methods on 116 and 117, could you not just add `transferManager` and  `asyncClient` as default args here?

Reply:
The zero argument one on line 116 can be replaced with default arguments so I've removed that one. I'll keep the one on line 118 as that lets us do `DAS3Client(s3AsyncClient)` while still being able to do `DAS3Client(transferManager)` which I can't do with the default argument one.